src/library/scala/collection/generic/DefaultSerializationProxy.scala

```diff
@@ -67,3 +67,21 @@ final class DefaultSerializationProxy[A](factory: Factory[A, Any], @transient pr
 
 @SerialVersionUID(3L)
 private[collection] case object SerializeEnd
+
+/** Mix-in trait to enable DefaultSerializationProxy for the standard collection types. Depending on the type
+  * it is mixed into, it will dynamically choose `iterableFactory`, `mapFactory`, `sortedIterableFactory` or
+  * `sortedMapFactory` for deserialization into the respective `CC` type. Override `writeReplace` or implement
+  * it directly without using this trait if you need a non-standard factory or if you want to use a different
+  * serialization scheme.
+  */
+trait DefaultSerializable extends Serializable { this: scala.collection.Iterable[_] =>
+  protected[this] def writeReplace(): AnyRef = {
+    val f: Factory[Any, Any] = this match {
+      case it: scala.collection.SortedMap[_, _] => it.sortedMapFactory.sortedMapFactory[Any, Any](it.ordering.asInstanceOf[Ordering[Any]]).asInstanceOf[Factory[Any, Any]]
+      case it: scala.collection.Map[_, _] => it.mapFactory.mapFactory[Any, Any].asInstanceOf[Factory[Any, Any]]
+      case it: scala.collection.SortedSet[_] => it.sortedIterableFactory.evidenceIterableFactory[Any](it.ordering.asInstanceOf[Ordering[Any]])
+      case it => it.iterableFactory.iterableFactory
+    }
+    new DefaultSerializationProxy(f, this)
+  }
+}
```

Comment:
If I understand correctly, developers who want to implement a custom collection type can mix this trait to support serialization?

~~~ scala
class MyCollection[A]
  extends Iterable[A]
    with IterableOps[A, MyCollection, MyCollection[A]]
    with DefaultSerializable { … }
~~~

Reply:
Yes, that's what most of the collection classes are doing now (except from some wrapper classes like the Java/Scala converters which use the default serialization)